app/controllers/profiles_controller.rb

```diff
@@ -57,24 +56,20 @@ def dashboard
       calculate_avg_rating
 
       @passengers = Passenger.where(journey_id: Journey.where(car: @car))
-
-      end
```

Comment:
Yes, it is correct. The code worked also before but now is better, as a part of the code is not run now if the teacher is the current user.